test/cmdlineTests/exp_base_literal/output

```diff
@@ -0,0 +1,238 @@
+                    value1 := abi_decode_t_uint256(add(headStart, offset), dataEnd)
+                }
+
+                {
+                    let offset := 64
+                    value2 := abi_decode_t_uint256(add(headStart, offset), dataEnd)
+                }
+
+                {
+                    let offset := 96
+                    value3 := abi_decode_t_uint256(add(headStart, offset), dataEnd)
+                }
+
+            }
+
+            function abi_encode_t_int256_to_t_int256_fromStack(value, pos) {
+                mstore(pos, cleanup_t_int256(value))
+            }
+
+            function abi_encode_t_uint256_to_t_uint256_fromStack(value, pos) {
+                mstore(pos, cleanup_t_uint256(value))
+            }
+
+            function abi_encode_tuple_t_uint256_t_int256_t_uint256_t_uint256__to_t_uint256_t_int256_t_uint256_t_uint256__fromStack(headStart , value0, value1, value2, value3) -> tail {
+                tail := add(headStart, 128)
+
+                abi_encode_t_uint256_to_t_uint256_fromStack(value0,  add(headStart, 0))
+
+                abi_encode_t_int256_to_t_int256_fromStack(value1,  add(headStart, 32))
+
+                abi_encode_t_uint256_to_t_uint256_fromStack(value2,  add(headStart, 64))
+
+                abi_encode_t_uint256_to_t_uint256_fromStack(value3,  add(headStart, 96))
+
+            }
+
+            function allocateMemory(size) -> memPtr {
+                memPtr := mload(64)
+                let newFreePtr := add(memPtr, size)
+                // protect against overflow
+                if or(gt(newFreePtr, 0xffffffffffffffff), lt(newFreePtr, memPtr)) { revert(0, 0) }
+                mstore(64, newFreePtr)
+            }
+
+            function checked_exp_t_rational_10_by_1_t_uint256(exponent) -> power {
+                exponent := cleanup_t_uint256(exponent)
+                if gt(exponent, 77) { revert(0, 0) }
+                power := exp(10, exponent)
+            }
+
+            function checked_exp_t_rational_115792089237316195423570985008687907853269984665640564039457584007913129639935_by_1_t_uint256(exponent) -> power {
+                exponent := cleanup_t_uint256(exponent)
+                if gt(exponent, 1) { revert(0, 0) }
+                power := exp(115792089237316195423570985008687907853269984665640564039457584007913129639935, exponent)
+            }
+
+            function checked_exp_t_rational_2_by_1_t_uint256(exponent) -> power {
+                exponent := cleanup_t_uint256(exponent)
+                if gt(exponent, 255) { revert(0, 0) }
+                power := exp(2, exponent)
+            }
+
+            function checked_exp_t_rational_minus_2_by_1_t_uint256(exponent) -> power {
+                exponent := cleanup_t_uint256(exponent)
+                if gt(exponent, 255) { revert(0, 0) }
+                power := exp(115792089237316195423570985008687907853269984665640564039457584007913129639934, exponent)
+            }
+
```

Comment:
@leonardoalt I added a commandlinetest and here's the generated function.

Reply:
Nice!